Userland/Libraries/LibCore/System.cpp

```diff
@@ -1273,19 +1274,14 @@ ErrorOr<void> adjtime(const struct timeval* delta, struct timeval* old_delta)
 ErrorOr<void> exec_command(Vector<StringView>& command, bool preserve_env)
 {
     Vector<StringView> exec_environment;
-    for (size_t i = 0; environ[i]; ++i) {
-        StringView env_view { environ[i], strlen(environ[i]) };
-        auto maybe_needle = env_view.find('=');
-
-        if (!maybe_needle.has_value())
-            continue;
-
+    Environment::for_each_entry([&](Environment::Entry& entry) {
```

Comment:
Yeah I really don't like how you can't tell what you're allowed to do with this Entry.
One might expect that modifying this object would modify the environment too, not a good thing.